pkg/apis/networking/validation/validation.go

```diff
@@ -496,29 +503,43 @@ func validateIngressBackend(backend *networking.IngressBackend, fldPath *field.P
 // IngressClass name.
 var ValidateIngressClassName = apimachineryvalidation.NameIsDNSSubdomain
 
-// ValidateIngressClass ensures that IngressClass resources are valid.
-func ValidateIngressClass(ingressClass *networking.IngressClass) field.ErrorList {
-	allErrs := apivalidation.ValidateObjectMeta(&ingressClass.ObjectMeta, false, ValidateIngressClassName, field.NewPath("metadata"))
-	allErrs = append(allErrs, validateIngressClassSpec(&ingressClass.Spec, field.NewPath("spec"))...)
-	return allErrs
+// ValidateIngressClassCreate ensures that IngressClass resources are valid as they are created.
+func ValidateIngressClassCreate(ingressClass *networking.IngressClass) field.ErrorList {
+	allowScope := utilfeature.DefaultFeatureGate.Enabled(features.IngressClassNamespacedParams)
```

Comment:
Although I know this has been done in validation (including by me), I think the proper way to do this is in strategy like what NetworkPolicy is doing (https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/networking/networkpolicy/strategy.go#L52)

Reply:
@robscott I tried taking up the strategy route but ran into a problem in validation code:
Scope is a new field *with* a default value. If Scope is force reset in strategy and validation package performs validation on the field, then the validation fails because Scope is an empty string.
We can update the validation code to run only if Scope is not empty but that doesn't seem correct.

Reply:
I went ahead and updated the code as per #97058 with one exception that validation for Scope field is feature gated. Does that seem reasonable?